tdrs-backend/tdpservice/parsers/schema_defs/tanf/t6.py

```diff
@@ -0,0 +1,171 @@
+"""Schema for HEADER row of all submission types."""
+
+
+from ...util import SchemaManager
+from ...transforms import calendar_quarter_to_rpt_month_year
+from ...fields import Field, TransformField
+from ...row_schema import RowSchema
+from ... import validators
+from tdpservice.search_indexes.models.tanf import TANF_T6
+
+
+s1 = RowSchema(
+    model=TANF_T6,
+    preparsing_validators=[
+        validators.hasLength(379),
+    ],
+    postparsing_validators=[
+        validators.sumIsEqual("NUM_APPLICATIONS", ["NUM_APPROVED", "NUM_DENIED"]),
+        validators.sumIsEqual("NUM_FAMILIES", ["NUM_2_PARENTS", "NUM_1_PARENTS", "NUM_NO_PARENTS"]),
+        validators.sumIsEqual("NUM_RECIPIENTS", ["NUM_ADULT_RECIPIENTS", "NUM_CHILD_RECIPIENTS"]),
+    ],
+    fields=[
+        Field(item="0", name='RecordType', type='string', startIndex=0, endIndex=2,
+              required=True, validators=[]),
+        Field(item="3", name='CALENDAR_QUARTER', type='number', startIndex=2, endIndex=7,
+              required=True, validators=[validators.dateYearIsLargerThan(1998),
+                                         validators.quarterIsValid()]),
+        TransformField(calendar_quarter_to_rpt_month_year(0), item="4", name='RPT_MONTH_YEAR', type='number',
+                       startIndex=2, endIndex=7, required=True, validators=[validators.dateYearIsLargerThan(1998),
+                                                                            validators.dateMonthIsValid()]),
+        Field(item="4A", name='NUM_APPLICATIONS', type='number', startIndex=7, endIndex=15,
+              required=True, validators=[validators.isInLimits(0, 99999999)]),
```

Comment:
Hmm, can we improve this `isInLimits(0, 999999)` call? I see we have various fields of different lengths too. I feel like we just need to ensure it's numeric and of a certain length. Do we already have a validator like that? Maybe just two validators instead of this sort of awkward one? The "magic number" 999999* is a code-smell.